src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const stateArray = [];
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'clear':
+        for (const key in stateClone) {
```

Comment:
you can declare `let stateClone = { ...state }` and for `case 'clear'` just do `stateClone = {}` (create an empty object).

Bu your solution is workable too.